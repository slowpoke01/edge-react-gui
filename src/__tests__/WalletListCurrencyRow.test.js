/* globals describe it expect */
/* eslint-disable flowtype/require-valid-file-annotation */

import * as React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'

import { getTheme } from '../components/services/ThemeContext.js'
import { WalletListRowComponent as Request } from '../components/themed/WalletListCurrencyRow.js'

describe('Request', () => {
  it('should render with loading props', () => {
    const renderer = new ShallowRenderer()

    const props = {
      currencyCode: 'string',
      gradient: true,
      onPress: () => undefined,
      onLongPress: () => undefined,
      showRate: true,
      paddingRem: 11 | [11],
      walletId: 'string',
      // eslint-disable-next-line react/no-unused-prop-types
      walletName: 'string',

      cryptoAmount: 'string',
      fiatBalanceSymbol: 'string',
      fiatBalanceString: 'string',
      walletNameString: 'string',
      exchangeRate: 'string',
      exchangeRates: 'GuiExchangeRates',
      fiatExchangeRate: 'string',
      walletFiatSymbol: 'string',
      theme: getTheme()
    }
    const actual = renderer.render(<Request {...props} />)

    expect(actual).toMatchSnapshot()
  })
})
