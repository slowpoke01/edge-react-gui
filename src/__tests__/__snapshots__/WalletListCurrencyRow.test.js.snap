// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request should render with loading props 1`] = `
<WithTheme(ClickableRowComponent)
  gradient={true}
  highlight={true}
  onLongPress={[Function]}
  onPress={[Function]}
  paddingRem={11}
>
  <WithTheme(CardContentComponent)
    image={
      <Memo(Connect(WithTheme(WalletProgressIconComponent)))
        currencyCode="string"
        walletId="string"
      />
    }
    subTitle="string"
    subValue="stringstring"
    title={
      <React.Fragment>
        <WithTheme(EdgeTextComponent)
          style={
            Object {
              "flexShrink": 1,
              "fontFamily": "Quicksand-Medium",
              "marginRight": 17,
            }
          }
        >
          string
        </WithTheme(EdgeTextComponent)>
        <WithTheme(EdgeTextComponent)
          style={
            Array [
              Object {
                "flexShrink": 2,
              },
              Object {
                "color": "#A4C7DF",
              },
            ]
          }
        >
          string str,ing.00 
        </WithTheme(EdgeTextComponent)>
      </React.Fragment>
    }
    value={
      <WithTheme(EdgeTextComponent)
        style={
          Object {
            "marginLeft": 11,
            "textAlign": "right",
          }
        }
      >
        string
      </WithTheme(EdgeTextComponent)>
    }
  />
</WithTheme(ClickableRowComponent)>
`;
